internal/state_test.go

```diff
@@ -7,19 +7,34 @@ import (
 )
 
 func TestBuildMetaUrl(t *testing.T) {
-	assert.Equal(t, buildMetadataURL("engine", "appId"), "http://engine:9076/v1/apps/appId/data/metadata")
+	assert.Equal(t, buildMetadataURL("engine", "appId"), "http://engine/v1/apps/appId/data/metadata")
 	assert.Equal(t, buildMetadataURL("engine:1234", "appId"), "http://engine:1234/v1/apps/appId/data/metadata")
-	assert.Equal(t, buildMetadataURL("wss://engine", "appId"), "https://engine:9076/v1/apps/appId/data/metadata")
-	assert.Equal(t, buildMetadataURL("ws://engine", "appId"), "http://engine:9076/v1/apps/appId/data/metadata")
+	assert.Equal(t, buildMetadataURL("wss://engine", "appId"), "https://engine/v1/apps/appId/data/metadata")
+	assert.Equal(t, buildMetadataURL("ws://engine", "appId"), "http://engine/v1/apps/appId/data/metadata")
 	assert.Equal(t, buildMetadataURL("wss://engine:1234", "appId"), "https://engine:1234/v1/apps/appId/data/metadata")
 	assert.Equal(t, buildMetadataURL("ws://engine:1234", "appId"), "http://engine:1234/v1/apps/appId/data/metadata")
 }
 
 func TestBuildEngineUrl(t *testing.T) {
```

Comment:
Looks good! Maybe add a test case without websocket prefix but with path and see what happens.

`engine/app/hello.qvf`

Should become: `ws://engine/app/hello.qvf` with the parsed app name `hello.qvf`